Formula/n/nodenv.rb

```diff
@@ -23,29 +23,38 @@ class Nodenv < Formula
   depends_on "node-build"
 
   def install
-    inreplace "libexec/nodenv" do |s|
-      s.gsub! "/usr/local", HOMEBREW_PREFIX
-      s.gsub! '"${BASH_SOURCE%/*}"/../libexec', libexec
-    end
-
-    %w[--version hooks versions].each do |cmd|
-      inreplace "libexec/nodenv-#{cmd}", "${BASH_SOURCE%/*}", libexec
-    end
-
-    # Compile bash extension
-    system "src/configure"
-    system "make", "-C", "src"
+    inreplace "libexec/nodenv", "/usr/local", HOMEBREW_PREFIX
 
     if build.head?
       # Record exact git revision for `nodenv --version` output
       inreplace "libexec/nodenv---version", /^(version=.+)/,
                                            "\\1--g#{Utils.git_short_head}"
     end
 
-    prefix.install "bin", "completions", "libexec"
+    prefix.install ["bin", "libexec", "nodenv.d"]
+    bash_completion.install "completions/nodenv.bash"
+    zsh_completion.install "completions/_nodenv"
+    man1.install "share/man/man1/nodenv.1"
   end
 
   test do
-    shell_output("eval \"$(#{bin}/nodenv init -)\" && nodenv --version")
+    # Create a fake node version and executable.
+    nodenv_root = Pathname(shell_output("#{bin}/nodenv root").strip)
+    node_bin = nodenv_root/"versions/1.2.3/bin"
+    foo_script = node_bin/"foo"
+    foo_script.write "echo hello"
+    chmod "+x", foo_script
+
+    # Test versions. The second `nodenv` call is a shell function; do not add a `bin` prefix.
+    versions = shell_output("eval \"$(#{bin}/nodenv init -)\" && nodenv versions").split("\n")
+    assert_equal 2, versions.length
+    assert_match(/\* system/, versions[0])
+    assert_equal("  1.2.3", versions[1])
```

Comment:
This needs adjustment, I think, since there is no `system` node (unlike Ruby).

If you prefer to keep this part of the test from `rbenv` unchanged, then doing something like
```ruby
depends_on "node" => :test
```
might help.

Reply:
Oh right, nice catch. (I must admit, I paid more attention to the formula body than the test when comparing to rbenv's formula.)